addons/mail/static/src/xml/activity_view.xml

```diff
@@ -1,54 +1,87 @@
                         <i class="fa fa-ellipsis-v fa-fw" data-toggle="dropdown"/>
                         <div class="dropdown-menu">
-                            <div t-foreach="type[2]" t-as="template" title="This action will send an email." class="o_template_element o_send_mail_template" t-att-data-template-id="template.id">
-                                <i class="fa fa-envelope fa-fw"/> <t t-esc="template.name"/>
-                            </div>
+                            <t t-foreach="type[2]" t-as="template" t-key="template.id">
+                                <div title="This action will send an email."
+                                     class="o_template_element o_send_mail_template"
+                                     t-att-data-activity-type-id="type[0]"
+                                     t-att-data-template-id="template.id"
+                                     t-on-click="_onSendMailTemplateClicked">
+                                    <i class="fa fa-envelope fa-fw"/> <t t-esc="template.name"/>
+                                </div>
+                            </t>
                         </div>
                     </span>
                 </div>
-                <div t-if="activityTypeIDs.length and !_.contains(activityTypeIDs, type[0])" class="mt24"/>
+                <KanbanColumnProgressBarAdapter Component="widgetComponents.KanbanColumnProgressBar"
+                    widgetArgs="[getProgressBarOptions(type[0]), getProgressBarColumnState(type[0])]"
+                    t-if="activityTypeIds.includes(type[0])"
+                    t-on-set-progress-bar-state="_onSetProgressBarState"/>
+                <div t-else="" class="mt24"/>
             </th>
         </tr>
     </thead>
 </t>
 
-<t t-name="mail.ActivityViewRow">
-    <td t-att-data-res-id="resId" t-att-data-activity-type-id="activityTypeId"
-        t-attf-class="o_activity_summary_cell #{activityGroup.state ? activityGroup.state : 'o_activity_empty_cell'} #{_.contains(widget.filteredResIDs, resId) ? 'o_activity_filter_' + widget.activeFilter : ''}">
-        <i t-if="!activityGroup.state" title="Create" class="text-center fa fa-plus"/>
+<t t-name="mail.ActivityViewBody" owl="1">
+    <tbody>
+        <tr t-foreach="activityResIds" t-as="resId" t-key="resId" class="o_data_row" t-att-data-res-id="resId">
+            <t t-set="record" t-value="props.data.find(data => data.res_id === resId)"/>
+            <td t-attf-class="{{ activeFilter.resIds.includes(resId) ? 'o_activity_filter_' + activeFilter.state : '' }}">
+                <ActivityRecordAdapter Component="widgetComponents.ActivityRecord"
+                    widgetArgs="[record, { qweb: qweb }]"/>
+            </td>
+            <t t-foreach="props.activity_types" t-as="type" t-key="type[0]">
+                <t t-call="mail.ActivityViewRow"/>
+            </t>
+        </tr>
+    </tbody>
+</t>
+
+<t t-name="mail.ActivityViewRow" owl="1">
```

Comment:
mmmmh I tought the row would be the tr, which makes more sense imho.

Reply:
I added mail.ActivityViewCell

The arch is now:
mail.ActivityViewBody (tbody)
  > mail.ActivityViewRow (1 tr)
    > mail.ActivityViewCell (1 td)

So we now could change the row independently from the cell

Reply:
:+1